pkg/provisioner/ironic/ironic.go

```diff
@@ -1094,12 +1087,22 @@ func (p *ironicProvisioner) Provision(hostConf provisioner.HostConfigData) (resu
 			return result, nil
 		}
 		p.log.Info("recovering from previous failure")
-		return p.startProvisioning(ironicNode, hostConf)
+		if provResult, err := p.startProvisioning(ironicNode, hostConf); err != nil || provResult.Dirty || provResult.ErrorMessage != "" {
+			return provResult, err
+		}
+
+		return p.changeNodeProvisionState(ironicNode,
+			nodes.ProvisionStateOpts{Target: nodes.TargetActive})
```

Comment:
This code used to be at the end of `startProvisioning()` but it has moved here because we only wanted to call it in the DeployFail state and not the Available state (which now calls `startProvisioning` - it used to be Manageable).